### ISSUE TITLE
Functional test failing frequently (but not always)

### ISSUE THREAD

**Issue by gkc** [atsign-foundation/at_client_sdk#994]
### Describe the bug

tests/at_functional_test/test/atclient_sync_callback_test.dart fails frequently at line 180 in the _'Verifying sync progress - server ahead'_ test - see https://github.com/atsign-foundation/at_client_sdk/actions/runs/4688763280/jobs/8309662564 for an example

### Steps to reproduce

Tests run; frequently this test fails (but not always)

### Expected behavior

Tests should reliably succeed or fail

**Comment by sitaram-kalluri** [comment]
Experienced the same issue. Ran the test with the log level set to the finest level and below are the observations.

In the previous test, the serverCommitId is 579 and in the failing test, when the update command is executed, the server commit id returned is 578. 

```
// ServerCommitId in the previous test
---------------------------------------------------
INFO|2023-04-11 04:39:00.091454|SyncService (@alice🛠)|Returning the serverCommitId 579 
.
.
.
// ServerCommitId in the failing test when update command is executed
 FINER|2023-04-11 04:38:55.129861|OutboundMessageListener|RECEIVED data:578 
```
 
Log Snippet
```
FINER|2023-04-11 04:39:00.081995|OutboundMessageListener|RECEIVED data: [{"id":"3","name":"lastCommitID","value":"579"}] 

INFO|2023-04-11 04:39:00.091454|SyncService (@alice🛠)|Returning the serverCommitId 579 

FINER|2023-04-11 04:39:00.091557|CommitLogKeyStore|Returning the lastSyncedEntry matching regex .* from cache. lastSyncedKey : public:fb_username-962ae781-2a51-4a78-8557-252a8e7daaff@alice🛠 with commitId 578 

FINER|2023-04-11 04:39:00.091686|SyncService (@alice🛠)|Informing 2 listeners of SyncProgress{atSign: @alice🛠, syncStatus: SyncStatus.success, isInitialSync: false, startedAt: 2023-04-11 04:39:00.002366Z, completedAt: 2023-04-11 04:39:00.091619Z, message: null, keyInfoList:[KeyInfo{key: public:fb_username-962ae781-2a51-4a78-8557-252a8e7daaff@alice🛠, syncDirection: SyncDirection.remoteToLocal , conflictInfo: null, commitOp: CommitOp.UPDATE}], localCommitIdBeforeSync:577, localCommitId:578, serverCommitId:579} 

❌ test/atclient_sync_callback_test.dart: Verifying sync progress - server ahead (failed)
  Setting encryption private key: true
  Setting encryption public key: true
  Setting self encryption key: true
  FINER|2023-04-11 04:38:55.080888|OutboundMessageListener|RECEIVED data:_dac5fc8a-4a4d-4de2-948f-7abbb0d7588e@alice🛠:afe2fd37-0e1c-47cc-bbcc-891dd989b1fd 
  
  FINER|2023-04-11 04:38:55.119789|OutboundMessageListener|RECEIVED data:success 
  
  FINER|2023-04-11 04:38:55.129861|OutboundMessageListener|RECEIVED data:578 
  
  INFO|2023-04-11 04:39:00.092104|FunctionalTestSyncService|Sync process completed. Sync Status: SyncStatus.success 
  
  Expected: <579>
    Actual: <578>
```

The complete log can found in : https://github.com/atsign-foundation/at_client_sdk/actions/runs/4663902943/jobs/8255652178

**Comment by gkc** [comment]
This has happened repeatedly this morning for the same PR. https://github.com/atsign-foundation/at_client_sdk/actions/runs/4760296464 currently on 5th attempt

I'm going to start looking at a fix for this; @sitaram-kalluri let me know if you're already on it please

**Comment by cpswan** [comment]
Also https://github.com/atsign-foundation/at_client_sdk/actions/runs/4781383676/jobs/8503562063?pr=1003

**Comment by purnimavenkatasubbu** [comment]
This issue has now been fixed with #1017
